ls1intum/Artemis
# `Exam mode`: Improve loading animation for assessment buttons

### Checklist
#### General


#### Client

### Motivation and Context


This pull requests removes unwanted animation when evaluating quizzes or assessing unsubmitted exams or toggling second correction in exam mode #4620

### Description

- Instead of using `isLoading` in `exam-assessment-buttons.component.ts` for the three different functionalities (loading, evaluating Quizzes, assess Exercises), we now have `isEvaluatingQuizExercises` and `isAssessingUnsubmittedExams` as additional boolean.
- `isTogglingSecondCorrection`as Map in `assessment-dashboard.component.ts`

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with at least 2 Exercises (one Quiz required)

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Course
4. Create an Exam
5. Create Quiz Exercise and for example a Text Exercise
7. Open Assessment Dashboard of the Exam
8. Press "Evaluate quizzes" and/or "Assess unsubmitted exams" -> loading animations should be visible only on the pressed button
9. Press on one of the two "Enable second correction" buttons -> loading animations should be visible only on the pressed button

### Review Progress





#### Code Review
#### Manual Tests
